src/backend/drm/legacy/surface.rs

```diff
@@ -132,13 +132,13 @@ impl<A: AsRawFd + 'static> Surface for LegacyDrmSurfaceInternal<A> {
     fn use_mode(&self, mode: Option<Mode>) -> Result<()> {
         let mut pending = self.pending.write().unwrap();
 
+        let error = || ErrorKind::DrmDev(format!("Error loading connector info on {:?}", self.dev_path()));
```

Comment:
It was the [about not having a closure with a block in a condition](https://rust-lang.github.io/rust-clippy/master/index.html#block_in_if_condition_stmt), but when i removed the block of it, `cargo fmt` would add it again so I thought I'd just make a let binding like how clippy suggests.

Reply:
In that case, I think it'd be more readable to reorganize it as follows, rather than split the error closure out.

```rust
let modes = connector::Into::load_from_device(..).chain_err(...)?.modes();
if !modes.contains(&mode) {
    ...
}
```
```

Reply:
Okay, I rearranged it a little differently, because the borrow checker didn't like it as a temporary value is freed with your suggestion.